R/col_align.R

```diff
@@ -1,95 +1,91 @@
-#' Align a column on character
+#' Apply alignment to a column
 #'
 #' @param col Character vector containing data values
-#' @param char_val Vector of one or more characters to align on. If NULL, data values will be aligned on the first occurrence of a decimal place or space. If more than one
+#' @param align Vector of one or more characters to align on. If NULL, data values will be aligned on the first occurrence of a decimal place or space. If more than one
 #' character is provided, alignment will be based on the first occurrence of any of the characters. For alignment based on white space, leading white spaces will be ignored.
 #'
 #' @return Character vector containing aligned data values
 #'
-#' @examples
-#'
-#' col <- c("xx.xx","  x", " xx (xx, xx)", "  x (xx.x, xx.x)", "xx, xx")
-#' tlang:::col_align_char(col, char_val = c(".", " ", ","))
-#'
 #' @importFrom tidyr separate replace_na
 #' @importFrom dplyr mutate across pull tibble
 #' @importFrom stringr str_dup str_c
-#'
-col_align_char <- function(col, char_val){
+#' @noRd
+apply_col_align <- function(col, align){
+
+  if (!all(align %in% c("left","right"))){
+    align <- paste(paste0("\\", align), collapse = "|")
+    align <- paste0("(?=[", align, "])")
+
+    tbl_dat <-  tibble(col = trimws(col)) %>%
+      separate(col, c("add_left","add_right"), sep = align, extra = "merge", fill = "right", remove = FALSE) %>%
+      mutate(across(c(.data$add_left, .data$add_right), function(x) {
+               replace_na(x, "") %>%
+                 nchar() %>%
+                 {max(.)-.} %>%
+                 {str_dup(" ", .)}
+             }))
+  } else {
+
+    tbl_dat <-  tibble(col = col) %>%
+      mutate(string_col = nchar(.data$col),
+             string_tot = max(.data$string_col),
+             space_to_add = str_dup(" ", .data$string_tot-.data$string_col)) %>%
+      rowwise %>%
+      mutate(add_left = ifelse(align=="left", "", .data$space_to_add),
+             add_right = ifelse(align=="right", "", .data$space_to_add))
 
-  if (is.null(char_val)){
-    char_val <- "."
   }
 
-  char_val <- paste(paste0("\\", char_val), collapse = "|")
-
-  char_val <- paste0("(?=[", char_val, "])")
-
-  tibble(col = trimws(col)) %>%
-    separate(col, c("string_left","string_right"), sep = char_val, extra = "merge", fill = "right", remove = FALSE) %>%
-    mutate(across(c(.data$string_left, .data$string_right), ~replace_na(.x, "") %>% nchar)) %>%
-    mutate(col_out = str_c(
-                        str_dup(" ", max(.data$string_left)-.data$string_left),
-                        col,
-                        str_dup(" ", max(.data$string_right)-.data$string_right))) %>%
+  tbl_dat %>%
+    rowwise %>%
+    mutate(col_out = str_c(.data$add_left,
+                           .data$col,
+                           .data$add_right)) %>%
     pull(.data$col_out)
```

Comment:
In a similar vein as my other comment by taking this out of a tibble into just vectors it will run faster cause you can just use str_c directly 
```suggestion
    str_c(tbl_dat$add_left, 
        tbl_dat$col,
        tbl_dat$add_right)
```